lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -0,0 +1,226 @@
+'use strict';
+class AwsCompileKafkaEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileKafkaEvents.bind(this),
+    };
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'kafka', {
+      type: 'object',
+      properties: {
+        accessConfigurations: {
+          type: 'array',
+          minItems: 1,
+          items: {
+            type: 'object',
+            minProperties: 1,
+            maxProperties: 1,
+            properties: {
+              basicAuth: {
+                type: 'array',
+                minItems: 1,
+                items: {
+                  type: 'string',
+                  pattern: secretsManagerArnRegex,
+                },
+              },
+              vpcSubnet: {
+                type: 'array',
+                minItems: 1,
+                items: {
+                  type: 'string',
+                  pattern: 'subnet-[a-z0-9]+',
+                },
+              },
+              vpcSecurityGroup: {
+                type: 'array',
+                minItems: 1,
+                items: {
+                  type: 'string',
+                  pattern: 'sg-[a-z0-9]+',
+                },
+              },
+              saslScram256Auth: {
+                type: 'array',
+                minItems: 1,
+                items: {
+                  type: 'string',
+                  pattern: secretsManagerArnRegex,
+                },
+              },
+              saslScram512Auth: {
+                type: 'array',
+                minItems: 1,
+                items: {
+                  type: 'string',
+                  pattern: secretsManagerArnRegex,
+                },
+              },
+            },
+          },
+        },
+        batchSize: {
+          type: 'number',
+          minimum: 1,
+          maximum: 10000,
+        },
+        enabled: {
+          type: 'boolean',
+        },
+        bootstrapServers: {
+          type: 'array',
+          minItems: 1,
+        },
+        startingPosition: {
+          type: 'string',
+          enum: ['LATEST', 'TRIM_HORIZON'],
+        },
+        topic: {
+          type: 'string',
+        },
+      },
+      additionalProperties: false,
+      required: ['accessConfigurations', 'bootstrapServers', 'topic'],
+    });
+  }
+
+  compileKafkaEvents() {
+    this.serverless.service.getAllFunctions().forEach((functionName) => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      // It is required to add the following statement in order to be able to connect to MSK cluster
```

Comment:
Just to confirm, is it still required at all times or only when VPC-based `accessConfiguration` is used?

Reply:
Hmm, not sure, but will double check.

Reply:
I believe it's only required for VPC-based `accessConfiguration`.